make clean build doesn't work out of the box

Hi @oleg-nenashev 

I want to try this project, because it seem to be very helpful

I follow the guide, and run _make clean build_ but I have a first error

```
$ make clean build
rm -rf tmp
java -jar .build/cwp-cli-1.5.jar \
	     -configPath packager-config.yml -version 256.0-test ""
 --batch-mode          : Enables the batch mode for the build (default: false)
 --bomPath FILE        : Path to the BOM file. If defined, it will override
                         settings in Config YAML
 --environment VAL     : Environment to be used
 --installArtifacts    : If set, the final artifacts will be automatically
                         installed to the local repository (current version -
                         only WAR) (default: false)
 -configPath FILE      : Path to the configuration YAML. See the tool's README
                         for format (default: packager-config.yml)
 -demo                 : Enables demo mode with predefined config file
                         (default: false)
 -mvnSettingsFile FILE : Path to a custom Maven settings file to be used within
                         the build
 -tmpDir FILE          : Temporary directory for generated files and the output
                         WAR. Defaults to 'tmp'
 -version VAL          : Version of WAR to be set. Defaults to '1.0-SNAPSHOT'
                         (default: 256.0-test)
Exception in thread "main" java.io.IOException: Failed to read command-line arguments
	at io.jenkins.tools.warpackager.cli.Main.main(Main.java:26)
Caused by: org.kohsuke.args4j.CmdLineException: No argument is allowed: 
	at org.kohsuke.args4j.CmdLineParser.parseArgument(CmdLineParser.java:509)
	at io.jenkins.tools.warpackager.cli.Main.main(Main.java:23)
make: *** [build] Error 1
```

To fix this I have delete _${CWP_OPTS}_ part of the build section in _Makefile_

So next I have the following error, but this time I can't fix it : 

```
Exception in thread "main" java.io.IOException: CasC section is declared, but CasC plugin is not declared in the plugins list
	at io.jenkins.tools.warpackager.lib.impl.Builder.verifyConfig(Builder.java:66)
	at io.jenkins.tools.warpackager.lib.impl.Builder.build(Builder.java:97)
	at io.jenkins.tools.warpackager.cli.Main.main(Main.java:56)
make: *** [build] Error 1
```
I put the full trace if needed
[build.txt](https://github.com/jenkinsci/ci.jenkins.io-runner/files/2988612/build.txt)

Do you know why ?

Thx you

So this is an issue with the cache:

```
INFO: Checking /home/onenashev/.m2/repository/io/jenkins/configuration-as-code/1.8/configuration-as-code-1.8.hpi
Mar 20, 2019 10:31:13 PM io.jenkins.tools.warpackager.lib.util.MavenHelper artifactExists
INFO: Dependency io.jenkins:configuration-as-code:1.8 was found in the non-HPI cache.  Delete /home/onenashev/.m2/cwp_non_hpi_cache/io/jenkins/configuration-as-code/1.8/configuration-as-code-1.8.hpi to attempt another resolution attempt.
Mar 20, 2019 10:31:13 PM io.jenkins.tools.warpackager.lib.config.Config processMavenDep
INFO: Skipping dependency, not an HPI file: io.jenkins:configuration-as-code:1.8
```

The cache is corrupted though https://repo.jenkins-ci.org/webapp/#/artifacts/browse/tree/General/releases/io/jenkins/configuration-as-code/1.8 looks OK


If I wipe the cache and re-run resolution, I get...

```
Mar 20, 2019 10:36:13 PM io.jenkins.tools.warpackager.lib.util.MavenHelper artifactExistsInLocalCache
INFO: Checking /home/onenashev/.m2/repository/io/jenkins/configuration-as-code/1.8/configuration-as-code-1.8.hpi
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:2.8:get (default-cli) on project standalone-pom: Couldn't download artifact: Failure to find io.jenkins:configuration-as-code:hpi:1.8 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced
[ERROR]
[ERROR] Try downloading the file manually from the project website.
[ERROR]
[ERROR] Then, install it using the command:
[ERROR]     mvn install:install-file -DgroupId=io.jenkins -DartifactId=configuration-as-code -Dversion=1.8 -Dpackaging=hpi -Dfile=/path/to/file
[ERROR]
[ERROR] Alternatively, if you host your own repository you can deploy the file there:
[ERROR]     mvn deploy:deploy-file -DgroupId=io.jenkins -DartifactId=configuration-as-code -Dversion=1.8 -Dpackaging=hpi -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
[ERROR]
[ERROR]
[ERROR]   io.jenkins:configuration-as-code:hpi:1.8
[ERROR]
[ERROR] from the specified remote repositories:
[ERROR]   central (https://repo.maven.apache.org/maven2, releases=true, snapshots=false)
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,9 @@
 Jenkins Config As Code Example.iml
 .build
+tmp/
+
+.factorypath
+.settings
+.project
+.classpath
+

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM jenkins/jenkins:2.138.3
+FROM jenkins/jenkins:2.164.1
 MAINTAINER Oleg Nenashev <o.v.nenashev@gmail.com>
 LABEL Description="This is a base image for a single-shot ci.jenkins.io demo" Vendor="Oleg Nenashev" Version="0.3"
 

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -19,7 +19,7 @@ node ("linux") {
         
         def makeCommand = "make build"
         if (hasSettingsXml) {
-            makeCommand += " -e CWP_OPTS='-mvnSettingsFile=${settingsXml}'"
+            makeCommand += " -e MVN_SETTINGS_FILE='${settingsXml}'"
         }
         infra.runWithMaven(makeCommand)
     }

--- a/Makefile
+++ b/Makefile
@@ -51,7 +51,7 @@ run:
 demo-plugin:
 	docker run --rm ${DOCKER_RUN_OPTS} \
 	    -v $(shell pwd)/demo/locale-plugin/:/workspace/ \
-	    $(DOCKER_TAG) ${DOCKER_RUN_OPTS}
+	    $(DOCKER_TAG)
 
 .PHONY: demo-plugin-local-lib
 demo-plugin-local-lib:

--- a/Makefile
+++ b/Makefile
@@ -7,8 +7,10 @@ CWP_MAVEN_REPO_PATH=io/jenkins/tools/custom-war-packager/custom-war-packager-cli
 CWP_VERSION=1.5
 DOCKER_TAG=jenkins4eval/ci.jenkins.io-runner:local-test
 PIPELINE_LIBRARY_DIR=/Users/nenashev/Documents/jenkins/infra/pipeline-library/
-CWP_OPTS=""
+CWP_OPTS=
 DOCKER_RUN_OPTS="-v maven-repo:/root/.m2"
+# It will not work properly if Jenkins repo is not set, so missing settings is not an option here
+MVN_SETTINGS_FILE ?= $(HOME)/.m2/settings.xml
 
 #TODO: Replace snapshot parsing by something more reliable
 ifneq (,$(findstring 1.6-2018,$(CWP_VERSION)))
@@ -36,7 +38,8 @@ docker:
 
 build: .build/cwp-cli-${CWP_VERSION}.jar
 	java -jar .build/cwp-cli-${CWP_VERSION}.jar \
-	     -configPath packager-config.yml -version ${VERSION} ${CWP_OPTS}
+	     -configPath packager-config.yml -version ${VERSION} ${CWP_OPTS} \
+		 -mvnSettingsFile ${MVN_SETTINGS_FILE}
 
 .PHONY: run
 run:

--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@ CWP_VERSION=1.5
 DOCKER_TAG=jenkins4eval/ci.jenkins.io-runner:local-test
 PIPELINE_LIBRARY_DIR=/Users/nenashev/Documents/jenkins/infra/pipeline-library/
 CWP_OPTS=
-DOCKER_RUN_OPTS="-v maven-repo:/root/.m2"
+DOCKER_RUN_OPTS=-v maven-repo:/root/.m2
 # It will not work properly if Jenkins repo is not set, so missing settings is not an option here
 MVN_SETTINGS_FILE ?= $(HOME)/.m2/settings.xml
 

--- a/README.md
+++ b/README.md
@@ -18,7 +18,8 @@ See the _Limitations_ section below for some of known limitations.
 
 ### Quickstart
 
-1. Checkout this repo
+0. Checkout this repo
+1. Configure Maven Settings file according to the [Jenkins Plugin Tutorial](https://wiki.jenkins.io/display/JENKINS/Plugin+tutorial) so that plugins can be properly downloaded for the pom.xml input
 2. Run `make docker` to build the base image
 3. Run `make clean build` to build the Jenkinsfile Runner image
 4. Run `make run` to run a simple demo

--- a/packager-config.yml
+++ b/packager-config.yml
@@ -15,7 +15,7 @@ buildSettings:
         noCache: true
       source:
         git: https://github.com/jenkinsci/jenkinsfile-runner.git
-        commit: 66626ef7c95c281d5b3ef9da9f5fdc36b4b42bf0
+        branch: 1.0-beta-7
     docker:
       base: "jenkins/ci.jenkins.io-runner.base"
       tag: "jenkins4eval/ci.jenkins.io-runner:local-test"
@@ -24,7 +24,7 @@ war:
   groupId: "org.jenkins-ci.main"
   artifactId: "jenkins-war"
   source:
-    version: "2.138.3"
+    version: "2.164.1"
 systemProperties:
   jenkins.model.Jenkins.slaveAgentPort: "50000"
   jenkins.model.Jenkins.slaveAgentPortEnforce: "true"

--- a/pom.xml
+++ b/pom.xml
@@ -38,6 +38,11 @@
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>jsch</artifactId>
+      <version>0.1.54.2</version>
+    </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>apache-httpcomponents-client-4-api</artifactId>
       <version>4.5.5-3.0</version>
     </dependency>

